week2/Q2.py

```diff
@@ -0,0 +1,47 @@
+from btree import BinaryTree
+from Q1 import get_ancestors
+
+
+def find_common_ancestor(ancestors1, ancestors2):
+	'''Finds lowest common ancestors given two lists of ancestors.
+
+	Args:
+		ancestors1: list, the ancestors of first given key (including the key)
+		ancestors2: list, the ancestors of second given key (including the key)
+
+	Returns:
+		element of a tree (type not specified)
+		None, if at least one of the keys is not present in the tree'''
+	pos1 = len(ancestors1) - 1
+	pos2 = len(ancestors2) - 1
+	lowest = None
+	while pos1 >= 0 and pos2 >= 0 and ancestors1[pos1] == ancestors2[pos2]:
```

Comment:
Optional: you can get done using a single counter that you update in the loop. Can you think of how?